docs/website/docs/dlt-ecosystem/verified-sources/filesystem/advanced.md

```diff
@@ -0,0 +1,176 @@
+
+```py
+files = filesystem(bucket_url="s3://my_bucket/data", file_glob="csv_folder/*.csv")
+pipeline.run(files)
+```
+
+The filesystem ensures consistent file representation across bucket types and offers methods to access and read data. You can quickly build pipelines to:
+
+- Extract text from PDFs ([unstructured data source](https://github.com/dlt-hub/verified-sources/tree/master/sources/unstructured_data)).
+- Stream large file content directly from buckets.
+- Copy files locally ([copy files](#copy-files-locally))
+
+### `FileItem` representation
+
+- All dlt sources/resources that yield files follow the [FileItem](https://github.com/dlt-hub/dlt/blob/devel/dlt/common/storages/fsspec_filesystem.py#L40) contract.
+- File content is typically not loaded (you can control it with the `extract_content` parameter of the filesystem resource). Instead, full file info and methods to access content are available.
+- Users can request an authenticated [fsspec AbstractFileSystem](https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/spec.html#AbstractFileSystem) instance.
+
+#### `FileItem` fields:
+
+- `file_url` - complete URL of the file (e.g. `s3://bucket-name/path/file`). This field serves as a primary key.
+- `file_name` - name of the file from the bucket URL.
+- `relative_path` - set when doing `glob`, is a relative path to a `bucket_url` argument.
+- `mime_type` - file's mime type. It is sourced from the bucket provider or inferred from its extension.
+- `modification_date` - file's last modification time (format: `pendulum.DateTime`).
+- `size_in_bytes` - file size.
+- `file_content` - content, provided upon request.
+
+:::info
+When using a nested or recursive glob pattern, `relative_path` will include the file's path relative to `bucket_url`. For instance, using the resource: `filesystem("az://dlt-ci-test-bucket/standard_source/samples", file_glob="met_csv/A801/*.csv")` will produce file names relative to the `/standard_source/samples` path, such as `met_csv/A801/A881_20230920.csv`. For local filesystems, POSIX paths (using "/" as separator) are returned.
+:::
+
+### File manipulation
+
+[FileItem](https://github.com/dlt-hub/dlt/blob/devel/dlt/common/storages/fsspec_filesystem.py#L40), backed by a dictionary implementation, offers these helper methods:
+
+- `read_bytes()` - method, which returns the file content as bytes.
+- `open()` - method which provides a file object when opened.
+- `filesystem` - field, which gives access to authorized `AbstractFilesystem` with standard fsspec methods.
+
+## Create Your Own Transformer
+
+While the `filesystem` resource yields the files from cloud storage or local filesystem, to get the actual records from the files you need to apply a transformer resource. `dlt` natively supports three file types: `csv`, `parquet`, and `jsonl` (more details in [filesystem transformer resource](../filesystem/basic#2-choose-the-right-transformer-resource)).
```

Comment:
Although the `filesystem` resource yields the files from cloud storage or **a** local filesystem,  you need to apply a transformer resource to retrieve the records from the files. `dlt` natively supports three file types: `csv`, `parquet`, and `jsonl` (more details in [filesystem transformer resource](../filesystem/basic#2-choose-the-right-transformer-resource)).